tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,272 @@
+            onInput={(e, value) => setSearch(value.toLowerCase())}
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Flex direction="column">
+            {filteredPrintables.map((val, index) => {
+              return (
+                <Flex.Item key={index}>
+                  <Box>
+                    <Flex direction="row">
+                      <Flex.Item grow>
+                        <Button
+                          fluid
+                          content={capitalize(val.name)}
+                          disabled={!val.can_make}
+                          color={val.hidden ? 'red' : null}
+                          onClick={() =>
+                            act('make', {
+                              index: val.index,
+                              multiplier: 1,
+                            })
+                          }>
+                          {' (' + // sorry for this shitcode, also yes this will break if an autolathe uses more than 2 material types
+                            (val.materials[Object.keys(materials)[0]] &&
+                            val.materials[Object.keys(materials)[1]]
+                              ? val.materials[Object.keys(materials)[0]] +
+                              ', ' +
+                              val.materials[Object.keys(materials)[1]]
+                              : val.materials[Object.keys(materials)[0]]
+                                ? val.materials[Object.keys(materials)[0]]
+                                : val.materials[Object.keys(materials)[1]]) +
+                            ') '}
+                        </Button>
+                      </Flex.Item>
+
+                      {(val.has_multipliers && (
+                        <Fragment>
+                          <Box width="2.5px" />
+                          <Flex.Item>
+                            <Flex direction="row">
+                              {Object.keys(val.multipliers).map((entry) => (
+                                <Flex.Item key={entry}>
+                                  <Button
+                                    content={'x' + entry}
+                                    onClick={() =>
+                                      act('make', {
+                                        index: val.index,
+                                        multiplier: entry,
+                                      })
+                                    }
+                                  />
+                                </Flex.Item>
+                              ))}
+                            </Flex>
+                          </Flex.Item>
+                        </Fragment>
+                      )) ||
+                        null}
+                    </Flex>
+                  </Box>
+                  <Box height="2.5px" />
+                </Flex.Item>
+              );
+            })}
+          </Flex>
+          <Box height="10px" />
+          <ElectricalPanel />
```

Comment:
This seems out of place in this component - I feel that it should be in the root component

Reply:
if it's not in here it ends up layering behind the list

Reply:
It's not the end of the world, I would say either the parent component neds to have its placement managed to refactor there or leave as is.